### ISSUE TITLE
Browser ESM build

### ISSUE THREAD

**Issue by e111077** [algolia/search-insights.js#356]
Hello, I'm trying to use this library and adding CJS is not acceptable currently in our build pipeline. Would it be possible to publish a modern browser ESM build for this library?

Thanks!

**Comment by kayluhb** [comment]
We're trying to implement this with vite and we're getting an issue with setting version even with configuring vite to execute with commonjs.

```js
  File: /Users/calebbrown/Projects/hydrogen-storefront/app/components/Analytics.tsx
TypeError: Cannot set properties of undefined (setting 'version')
    at /Users/calebbrown/Projects/hydrogen-storefront/node_modules/search-insights/index-browser.cjs.js:6:25
    at Array.forEach (<anonymous>)
    at /Users/calebbrown/Projects/hydrogen-storefront/node_modules/search-insights/index-browser.cjs.js:4:17
    at Object.runViteModule (Users/user/Projects/hydrogen-storefront/node_modules/@shopify/mini-oxygen/dist/vite/worker-entry.js:1181:11)
    at ViteRuntime.directRequest (Users/user/Projects/hydrogen-storefront/node_modules/@shopify/mini-oxygen/dist/vite/worker-entry.js:1026:60)
    at ViteRuntime.cachedRequest (Users/user/Projects/hydrogen-storefront/node_modules/@shopify/mini-oxygen/dist/vite/worker-entry.js:950:79)
    at /app/components/Analytics.tsx:3:31
    at Object.runViteModule (Users/user/Projects/hydrogen-storefront/node_modules/@shopify/mini-oxygen/dist/vite/worker-entry.js:1181:11)
    at ViteRuntime.directRequest (Users/user/Projects/hydrogen-storefront/node_modules/@shopify/mini-oxygen/dist/vite/worker-entry.js:1026:60)
    at ViteRuntime.cachedRequest (Users/user/Projects/hydrogen-storefront/node_modules/@shopify/mini-oxygen/dist/vite/worker-entry.js:950:79)
```

**Comment by JasonBerry** [comment]
Released in [2.16.0](https://github.com/algolia/search-insights.js/releases/tag/v2.16.0)
